plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1890,6 +1895,46 @@ private static void checkNoMetadataUpdates(DeltaLakeCommitSummary commitSummary)
         }
     }
 
+    private static void checkIfCommittedAddedFilesConflictWithCurrentOperation(TupleDomain<DeltaLakeColumnHandle> enforcedPartitionConstraint, IsolationLevel isolationLevel, DeltaLakeCommitSummary commitSummary)
+    {
+        List<AddFileEntry> addedFiles = commitSummary.getIsBlindAppend().orElse(false)
+                ? List.of() // Do not conflict with blind appends
+                : commitSummary.getAddedFiles();
+
+        if (addedFiles.isEmpty()) {
+            return;
+        }
+
+        boolean readWholeTable = enforcedPartitionConstraint.isAll();
+        if (readWholeTable) {
+            throw new TransactionFailedException("Conflicting concurrent writes found. Data files added in the modified table by concurrent write operation.");
+        }
+
+        Map<DeltaLakeColumnHandle, Domain> enforcedDomains = enforcedPartitionConstraint.getDomains().orElseThrow();
+        Optional<AddFileEntry> unreadAddAction = addedFiles.stream()
+                .filter(addAction -> partitionMatchesPredicate(addAction.getCanonicalPartitionValues(), enforcedDomains))
+                .findFirst();
+        if (unreadAddAction.isPresent()) {
+            throw new TransactionFailedException("Conflicting concurrent writes found. Data files were  added on the table with isolation level %s in the partition '%s' of the modified table  by concurrent write operation."
+                    .formatted(isolationLevel, unreadAddAction.get().getPartitionValues()));
+        }
+    }
+
+    private static void checkIfCommittedRemovedFilesConflictWithCurrentOperation(IsolationLevel isolationLevel, DeltaLakeCommitSummary commitSummary)
+    {
+        List<RemoveFileEntry> removedFiles = commitSummary.getIsBlindAppend().orElse(false)
```

Comment:
Can we have blind append when there are remove files?
Maybe we should always check `removedFiles.isEmpty` to be sure?